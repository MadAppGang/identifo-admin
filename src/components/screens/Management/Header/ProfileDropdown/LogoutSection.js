import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logout } from '~/modules/auth/actions';
import LogoutIcon from '~/components/shared/icons/LogoutIcon';

const LogoutSection = (props) => {
  return (
    <button
      type="button"
      className="iap-profile-dropdown__section"
      onClick={props.logout}
    >
      <LogoutIcon />
      <span>Logout</span>
    </button>
  );
};

LogoutSection.propTypes = {
  logout: PropTypes.func.isRequired,
};

export { LogoutSection };

export default connect(null, { logout })(LogoutSection);
