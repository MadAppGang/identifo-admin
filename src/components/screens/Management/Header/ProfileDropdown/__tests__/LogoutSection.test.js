import { LogoutSection } from '../LogoutSection';

describe('<LogoutSection />', () => {
  test('renders as expected', () => {
    expect(shallow(<LogoutSection logout={jest.fn()} />)).toMatchSnapshot();
  });

  test('calls logout prop function on button click', () => {
    const logout = jest.fn();

    shallow(<LogoutSection logout={logout} />).find('button').simulate('click');
    expect(logout).toHaveBeenCalled();
  });
});
