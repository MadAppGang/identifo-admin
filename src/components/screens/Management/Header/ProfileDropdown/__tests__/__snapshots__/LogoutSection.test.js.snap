// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutSection /> renders as expected 1`] = `
<button
  className="iap-profile-dropdown__section"
  onClick={[MockFunction]}
  type="button"
>
  <DropdownIcon />
  <span>
    Logout
  </span>
</button>
`;
